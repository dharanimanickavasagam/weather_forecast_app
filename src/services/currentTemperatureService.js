import http from "../httpAxiosModule/httpAxiosModule";
import url from "../config.json";

export const getCurrentTemperature = async (lat, lon) => {
  const { data } = await http.get(
    `${url.apiEndPoint}/weather?lat=${lat}&lon=${lon}&appid=${url.appid}`
  );
  return data;
};
