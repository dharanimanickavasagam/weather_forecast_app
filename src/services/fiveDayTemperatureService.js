import http from "../httpAxiosModule/httpAxiosModule";
import url from "../config.json";

export const getFiveDayTemperature = async (lat, lon) => {
  const { data } = await http.get(
    `${url.apiEndPoint}/forecast?lat=${lat}&lon=${lon}&appid=${url.appid}`
  );
  return data;
};

//test 4

//test 5
