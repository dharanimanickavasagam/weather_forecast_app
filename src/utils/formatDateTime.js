import moment from "moment";

export const formatDateTime = (dateTime, dateTimeFormat) => {
  return moment(dateTime).format(dateTimeFormat);
};
