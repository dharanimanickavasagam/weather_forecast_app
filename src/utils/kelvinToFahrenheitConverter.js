export const kelvinToFahrenheitConverter = kelvinTemperature => {
  return Math.round((kelvinTemperature - 273.15) * (9 / 5) + 32);
};
