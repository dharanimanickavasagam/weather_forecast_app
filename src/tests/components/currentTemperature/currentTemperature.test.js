import React from "react";
import { shallow, mount } from "enzyme";
import CurrentTemperature from "../../../components/currentTemperature/currentTemperature";
import {
  weatherDetails,
  main,
  weatherTableDetails,
  mockWeatherTableDetails
} from "../../fixures/currentTemperature";
import { kelvinToFahrenheitConverter } from "../../../utils/kelvinToFahrenheitConverter";
import { getCurrentTemperature } from "../../../services/currentTemperatureService";

let wrapper;

jest.mock("../../../utils/kelvinToFahrenheitConverter", () => {
  return { kelvinToFahrenheitConverter: jest.fn(() => 72) };
});

jest.mock("../../../utils/formatDateTime", () => {
  return { formatDateTime: jest.fn(() => "Tuesday, 07-02-2019 19:20") };
});

jest.mock("../../../services/currentTemperatureService", () => ({
  getCurrentTemperature: jest.fn(() =>
    Promise.resolve({
      main: {
        temp: 283.04,
        pressure: 1015,
        humidity: 76,
        temp_min: 279.82,
        temp_max: 285.37
      },
      weather: [
        { id: 800, main: "Clear", description: "clear sky", icon: "01n" }
      ],
      name: "Romeo"
    })
  )
}));

beforeAll(() => {
  window.alert = () => {};
  //leaving the commented ones for my reference
  /*   dateNowSpy = jest.spyOn(Date, "now").mockImplementation(() => 1562109736100);
  mockKelvinToFahrenheitConverter = jest.fn().mockImplementationOnce(() => 54);
  CurrentTemperature.prototype.getWeatherDetails = function() {
    return weatherTableDetails;
  }; */
});

beforeEach(() => {
  wrapper = shallow(<CurrentTemperature />);
});

describe("match snapshots with previous ", () => {
  test("match snapshot", () => {
    wrapper.setState({ isLoading: false });
    expect(wrapper).toMatchSnapshot();
  });
});

describe("it should not render the component when isLoading is true", () => {
  test("it should not render weather details when data is loading, isLoading:true", () => {
    wrapper.setState({ isLoading: true });
    expect(wrapper.find("Header")).not.toBe(true);
    expect(wrapper.find("Table")).not.toBe(true);

    expect(wrapper.find(".loading").text()).toBe("Loading...");
    expect(wrapper).toMatchSnapshot();
  });
});

describe("it renders component when isLoading is false", () => {
  test("it should render Header and Table after the data is loaded", () => {
    wrapper.setState({ isLoading: false });

    expect(wrapper.find("Header")).toBeTruthy();
    expect(wrapper.find("Table")).toBeTruthy();
  });

  test("it should render the current city location ", () => {
    wrapper.setState({ isLoading: false });
    wrapper.setState({ cityName: "Detroit" });
    expect(wrapper.find("Header").prop("name")).toEqual("Detroit");
  });

  test("it should render the current city location's weather details with mock data", () => {
    wrapper.setState({ isLoading: false });
    wrapper.setState({ weatherDetails, main });
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("img").prop("src")).toEqual(
      `http://openweathermap.org/img/wn/${weatherDetails.icon}@2x.png`
    );

    expect(wrapper.find(".flexContainer__weather").text()).toBe(`72Â°F`);

    expect(wrapper.find(".weatherDetails").text()).toBe(
      weatherDetails.description
    );

    expect(wrapper.find("Table").prop("data")).toStrictEqual(
      mockWeatherTableDetails
    );
  });
});

describe("Testing Component methods ", () => {
  test("testing getWeatherDetails() ", () => {
    const currentTemperature = new CurrentTemperature();
    const res = currentTemperature.getWeatherDetails(main);
    expect(kelvinToFahrenheitConverter).toHaveBeenCalled();

    expect(res).toStrictEqual(mockWeatherTableDetails);
  });

  test("testing getCurrentDateAndTime() ", () => {
    const currentTemperature = new CurrentTemperature();
    const res = currentTemperature.getCurrentDateAndTime();
    expect(res).toStrictEqual("Tuesday, 07-02-2019 19:20");
  });
});

describe("testing the method componentDidUpdate", () => {
  test("trigger a componentDidUpdate when the prevProps and the currentProps are different", async () => {
    const wrapper = mount(
      <CurrentTemperature latitude={null} longitude={null} />
    );
    wrapper.setProps({ latitude: 13, longitude: 12 });

    expect(getCurrentTemperature).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
