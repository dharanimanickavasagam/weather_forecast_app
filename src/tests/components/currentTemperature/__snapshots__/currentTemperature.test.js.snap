// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component when isLoading is false it should render the current city location's weather details with mock data 1`] = `
<main
  className="container"
>
  <Header
    name=""
    variant="h6"
  />
  <div
    className="flexContainer"
  >
    <div
      className="flexContainer__weatherIcon"
    >
      <img
        alt="weather_icon"
        src="http://openweathermap.org/img/wn/01n@2x.png"
      />
    </div>
    <div
      className="flexContainer__weather"
    >
      72
      °F
    </div>
  </div>
  <div
    className="weatherDetails"
  >
    Clear sky
  </div>
  <div
    className="dateTime"
  >
    Tuesday, 07-02-2019 19:20
  </div>
  <Table
    columns={
      Array [
        "Pressure (hpa)",
        "Humidity %",
        "Min °F",
        "Max °F",
      ]
    }
    data={
      Array [
        Object {
          "humidity": 93,
          "pressure": 1024,
          "temp_max": 72,
          "temp_min": 72,
        },
      ]
    }
  />
</main>
`;

exports[`it should not render the component when isLoading is true it should not render weather details when data is loading, isLoading:true 1`] = `
<main
  className="container"
>
  <div
    className="loading"
  >
    Loading...
  </div>
</main>
`;

exports[`match snapshots with previous  match snapshot 1`] = `
<main
  className="container"
>
  <Header
    name=""
    variant="h6"
  />
  <div
    className="flexContainer"
  >
    <div
      className="flexContainer__weatherIcon"
    >
      <img
        alt="weather_icon"
        src="http://openweathermap.org/img/wn/undefined@2x.png"
      />
    </div>
    <div
      className="flexContainer__weather"
    >
      72
      °F
    </div>
  </div>
  <div
    className="weatherDetails"
  />
  <div
    className="dateTime"
  >
    Tuesday, 07-02-2019 19:20
  </div>
  <Table
    columns={
      Array [
        "Pressure (hpa)",
        "Humidity %",
        "Min °F",
        "Max °F",
      ]
    }
    data={
      Array [
        Object {
          "temp_max": 72,
          "temp_min": 72,
        },
      ]
    }
  />
</main>
`;

exports[`testing the method componentDidUpdate trigger a componentDidUpdate when the prevProps and the currentProps are different 1`] = `
<main
  className="container"
>
  <div
    className="loading"
  >
    Loading...
  </div>
</main>
`;
